pageserver/src/billing_metrics.rs

```diff
@@ -32,16 +35,23 @@ use reqwest::Url;
 /// "tenant_id": "5d07d9ce9237c4cd845ea7918c0afa7d",
 /// "timeline_id": "00000000000000000000000000000000",
 /// "time": ...,
+/// "idempotency_key":
 /// "value": 12345454,
 /// }
 /// ```
+#[serde_as]
 #[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq, Ord, PartialOrd)]
 pub struct BillingMetric {
     pub metric: BillingMetricKind,
+    #[serde(rename = "type")]
     pub metric_type: &'static str,
```

Comment:
NIT: Having struct named Metric with fields prefixed by metric doesnt look natural to me. So to me it makes sense to rename field in the struct itself without serde rename

Reply:
unfortunately, type is a reserved keyword), so it can't be a field name..
Or did you mean something else?

Reply:
When `type` is a reserved keyword, it's usually written as `type_` or `ty`.

Reply:
Also reserved keywords can be still used with r#. See https://doc.rust-lang.org/beta/rust-by-example/compatibility/raw_identifiers.html

Reply:
Kind and Type are also quite similar in semantics. Should we replace them with something else? My suggestion:

absolute/delta -> enum Kind
remote_storate_size -> name or id

Wdyt?